### PR TITLE
Js set home on demote

### PR DESCRIPTION
**Pull request by jsingle**
@thieman I was having trouble with demoted boot2docker running from the plist, since the demoted process still has HOME set to `/var/root`

### CODE REVIEW

File: dusty/subprocess.py [hunk 1]
```diff
@@ -17,10 +17,15 @@
 def _demote_to_user(user_name):
     def _demote():
         pw_record = pwd.getpwnam(user_name)
+        _set_demoted_home_dir(user_name)
         os.setgid(pw_record.pw_gid)
         os.setuid(pw_record.pw_uid)
     return _demote
 
+def _set_demoted_home_dir(user_name):
+    home_dir = '/var/root' if user_name == 'root' else '/Users/{}'.format(user_name)
```
Comment (thieman) [inline dusty/subprocess.py]: This is the wrong way to do this. Use `os.path.expanduser`

**Comment by thieman** [comment]
This doesn't feel like the right way to do this. Are there alternatives?

File: dusty/subprocess.py (hunk 1 above)
Comment (jsingle) [inline dusty/subprocess.py]: Ah that's nifty

**Comment by thieman** [comment]
Well actually I guess this makes sense, just change it to use expanduser

**Comment by jsingle** [comment]
There might be a better way that we should be demoting - where we actually login as a user, then run the shell args

**Pull request merged**